crates/cli/src/app_builder.rs

```diff
@@ -99,22 +105,49 @@ pub trait AppBuilder: CommandBuilder {
         fs::create_dir_all(&output_dir)?;
         fs::create_dir_all(&param_dir)?;
 
+        let host_mode = Self::parse_host_mode(&top_matches);
+
         match top_matches.subcommand() {
-            Some(("setup", _)) => exec_setup::<ExecutionArg, DefaultHostEnvBuilder>(
-                zkwasm_k,
-                Self::AGGREGATE_K,
-                Self::NAME,
-                wasm_binary,
-                phantom_functions,
-                &output_dir,
-                &param_dir,
-            ),
-            Some(("checksum", _)) => exec_image_checksum::<ExecutionArg, DefaultHostEnvBuilder>(
-                zkwasm_k,
-                wasm_binary,
-                phantom_functions,
-                &output_dir,
-            ),
+            Some(("setup", _)) => match host_mode {
+                HostMode::DEFAULT => exec_setup::<ExecutionArg, DefaultHostEnvBuilder>(
```

Comment:
Why not use `StandardEnvBuilder` as default host? And I doubt that users can differentiate them by name.

Reply:
A better name might be standard --> default, default --> minimize, and then I can add a customize tag for those who would like to load env via a json file. But I think it can be done in a separate PR later.